core/src/main/scala/caliban/execution/Feature.scala

```diff
@@ -1,8 +1,62 @@
 package caliban.execution
 
-sealed trait Feature
+import caliban.introspection.adt.{ __Directive, __DirectiveLocation, __InputValue }
+import caliban.parsing.adt.Directives
+import caliban.schema.Types
+
+sealed trait Feature {
+  def index: Int
+  def mask: Int
+  def directives: List[__Directive]
+}
 
 object Feature {
-  case object Defer  extends Feature
-  case object Stream extends Feature
+  type Flags = Int
+
+  private def isEnabled(flags: Flags, feature: Feature): Boolean = (flags & feature.mask) != 0
+  def isDeferEnabled(flags: Flags): Boolean                      = isEnabled(flags, Defer)
+  def isStreamEnabled(flags: Flags): Boolean                     = isEnabled(flags, Stream)
```

Comment:
In this case it doesn't really matter since there are only 2 objects extending `Feature`, but to avoid the virtual call to `.mask` I usually do this:

```suggestion
  private def isEnabled(flags: Flags, mask: Int): Boolean = (flags & mask) != 0
  def isDeferEnabled(flags: Flags): Boolean                      = isEnabled(flags, Defer.mask)
  def isStreamEnabled(flags: Flags): Boolean                     = isEnabled(flags, Stream.mask)
```